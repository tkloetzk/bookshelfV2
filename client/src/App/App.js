import React from 'react'
import SwipeableViews from 'react-swipeable-views'
import Tab from '@material-ui/core/Tab'
import Tabs from '@material-ui/core/Tabs'
import { makeStyles } from '@material-ui/styles'
import Search from './Search/Search'
import Header from './header/Header'

const useStyles = makeStyles((theme) => ({
  tabs: {
    background: theme.palette.primary.gray,
  },
  container: {
    marginTop: 103,
  },
}))

export default function App() {
  const classes = useStyles()
  const [index, setIndex] = React.useState(0)

  return (
    <>
      <Header>Book Review Aggregator</Header>
      <div className={classes.container}>
        <Tabs
          value={index}
          indicatorColor="primary"
          textColor="primary"
          variant="fullWidth"
          className={classes.tabs}
        >
          <Tab
            label="Search"
            data-testid="searchTab"
            onClick={() => setIndex(0)}
          />
          <Tab
            label="Bookshelf"
            data-testid="bookshelfTab"
            onClick={() => setIndex(1)}
          />
        </Tabs>
        <SwipeableViews index={index} onChangeIndex={setIndex}>
          <Search />
          <div data-testid="bookshelfView">Test2</div>
        </SwipeableViews>
      </div>
    </>
  )
}
