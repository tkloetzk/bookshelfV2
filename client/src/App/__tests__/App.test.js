import { MuiThemeProvider } from '@material-ui/core/styles';
import React from 'react';
import { mount } from 'enzyme';
import muiTheme from '../../config/themeConfig';
import App from '../App';

describe('App', () => {
  let component;

  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation(init => [init, setState]);

  beforeEach(() => {
    component = mount(
      <MuiThemeProvider theme={muiTheme}>
        <App />
      </MuiThemeProvider>,
    ).find('App');
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('render', () => {
    it('should render as expected', () => {
      expect(component.html()).toMatchSnapshot();
    });
  });

  describe('handleChangeIndex', () => {
    it('updates the index when clicking the bookshelf tab', () => {
      component
        .find('#bookshelfTab')
        .hostNodes()
        .props()
        .onClick();
      expect(setState).toHaveBeenCalledWith(1);
    });
    it('updates the index when clicking the search tab', () => {
      component
        .find('#searchTab')
        .hostNodes()
        .props()
        .onClick();
      expect(setState).toHaveBeenCalledWith(0);
    });
  });
});
