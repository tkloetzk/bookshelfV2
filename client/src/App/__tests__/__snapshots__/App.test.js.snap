// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render should render as expected 1`] = `
<DocumentFragment>
  <header
    class="header"
    style="z-index: 1;"
  >
    <div
      class="header-content display-4"
    >
      Book Review Aggregator
    </div>
  </header>
  <div
    class="makeStyles-container-2"
  >
    <div
      class="MuiTabs-root makeStyles-tabs-1"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          class="MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected MuiTab-fullWidth"
            data-testid="searchTab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Search
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
            data-testid="bookshelfTab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Bookshelf
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-27 PrivateTabIndicator-colorPrimary-28 MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      style="overflow-x: hidden;"
    >
      <div
        class="react-swipeable-view-container"
        style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
      >
        <div
          aria-hidden="false"
          data-swipeable="true"
          style="width: 100%; flex-shrink: 0; overflow: auto;"
        >
          <div
            class="MuiGrid-root makeStyles-container-31 MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
            data-testid="searchView"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                    data-testid="searchBar"
                    rows="1"
                    style="height: 0px; overflow: hidden;"
                  />
                  <textarea
                    aria-hidden="true"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                    readonly=""
                    style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; width: 100%;"
                    tabindex="-1"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-173 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-174"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
                data-testid="searchButton"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Search
                </span>
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App tabs should show Bookshelf when bookshelf tab is clicked 1`] = `
<DocumentFragment>
  <header
    class="header"
    style="z-index: 1;"
  >
    <div
      class="header-content display-4"
    >
      Book Review Aggregator
    </div>
  </header>
  <div
    class="makeStyles-container-404"
  >
    <div
      class="MuiTabs-root makeStyles-tabs-403"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          class="MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
            data-testid="searchTab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Search
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected MuiTab-fullWidth"
            data-testid="bookshelfTab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Bookshelf
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-429 PrivateTabIndicator-colorPrimary-430 MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      style="overflow-x: hidden;"
    >
      <div
        class="react-swipeable-view-container"
        style="flex-direction: row; transition: transform 0.35s cubic-bezier(0.15, 0.3, 0.25, 1) 0s; direction: ltr; display: flex; will-change: transform; transform: translate(-100%, 0);"
      >
        <div
          aria-hidden="true"
          data-swipeable="true"
          style="width: 100%; flex-shrink: 0; overflow: auto;"
        >
          <div
            class="MuiGrid-root makeStyles-container-433 MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
            data-testid="searchView"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                    data-testid="searchBar"
                    rows="1"
                    style="height: 0px; overflow: hidden;"
                  />
                  <textarea
                    aria-hidden="true"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                    readonly=""
                    style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; width: 100%;"
                    tabindex="-1"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-575 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-576"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
                data-testid="searchButton"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Search
                </span>
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          data-swipeable="true"
          style="width: 100%; flex-shrink: 0; overflow: auto;"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App tabs should show Search when search tab is clicked 1`] = `
<DocumentFragment>
  <header
    class="header"
    style="z-index: 1;"
  >
    <div
      class="header-content display-4"
    >
      Book Review Aggregator
    </div>
  </header>
  <div
    class="makeStyles-container-203"
  >
    <div
      class="MuiTabs-root makeStyles-tabs-202"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          class="MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected MuiTab-fullWidth"
            data-testid="searchTab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Search
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
            data-testid="bookshelfTab"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Bookshelf
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-228 PrivateTabIndicator-colorPrimary-229 MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      style="overflow-x: hidden;"
    >
      <div
        class="react-swipeable-view-container"
        style="flex-direction: row; transition: transform 0.35s cubic-bezier(0.15, 0.3, 0.25, 1) 0s; direction: ltr; display: flex; will-change: transform; transform: translate(0%, 0);"
      >
        <div
          aria-hidden="false"
          data-swipeable="true"
          style="width: 100%; flex-shrink: 0; overflow: auto;"
        >
          <div
            class="MuiGrid-root makeStyles-container-232 MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
            data-testid="searchView"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                    data-testid="searchBar"
                    rows="1"
                    style="height: 0px; overflow: hidden;"
                  />
                  <textarea
                    aria-hidden="true"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                    readonly=""
                    style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; width: 100%;"
                    tabindex="-1"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-374 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-375"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
                data-testid="searchButton"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Search
                </span>
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          data-swipeable="true"
          style="width: 100%; flex-shrink: 0; overflow: auto;"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
